nav2_map_server/src/map_saver_cli.cpp

```diff
@@ -26,50 +26,57 @@ const char * USAGE_STRING{
   "  map_saver [--occ <threshold_occupied>] [--free <threshold_free>] [--fmt <image_format>] "
   "[--mode trinary/scale/raw] [-f <mapname>] [ROS remapping args]"};
 
-int main(int argc, char ** argv)
-{
+typedef enum {TYPE_STR, TYPE_INT} VAR_TYPE;
+struct cmd_struct {
+  const char *cmd;
+  const char *param_name;
+  VAR_TYPE type;
+};
+
+int main(int argc, char ** argv) {
+  const struct cmd_struct commands[] = {
+    { "-f", "output_file_no_ext", TYPE_STR },
+    { "--occ", "threshold_occupied", TYPE_INT },
+    { "--free", "threshold_free", TYPE_INT },
+    { "--mode", "map_mode", TYPE_STR },
+    { "--fmt", "image_format", TYPE_STR },
+  };
+
```

Comment:
How about checking if (argc >= 2) first? Otherwise, below code seems to be invalid access.
```cpp
std::vector<std::string> arguments(argv + 1, argv + argc);
```

Reply:
@DongheeYe  Thank you for your review. :)
I tried to remain the original source codes, but it will not have invalid access because it is just address bound. 
Also, argc can be 1. (no param)